src/print-results.ts

```diff
@@ -251,3 +269,37 @@ function sumCounts(results: ESLint.LintResult[]): TodoFormatterCounts {
 function pluralize(word: string, count: number): string {
   return count === 1 ? word : `${word}s`;
 }
+
+function filterTodos(
+  results: ESLint.LintResult[],
+  options: TodoPrintOptions
+): ESLint.LintResult[] {
+  return results.reduce((acc, result) => {
+    const messages = result.messages as TodoResultMessage[];
+    if (messages.length === 0) {
+      return acc;
+    }
+
+    if (options.includeTodo) {
+      return [...acc, result];
+    }
+
+    const areAllMessagesTodo = messages.every(
+      (message) => message.severity === Severity.todo
+    );
+
+    if (areAllMessagesTodo) {
+      return acc;
+    }
```

Comment:
The cost of doing this check sorta negates avoiding doing the filter below. I'd just remove it for simplification.

```suggestion
```

Reply:
@scalvert removing this will cause a `LintResult` with zero messages to be includes (if it's all todos). I _think_ it makes sense to keep this condition, but happy to defer to you.

That is, given the test fixture `todos-only.js`, if we remove this condition that file _will_ be included but the `messages` will be an empty array as we'll filter them all out below.

Reply:
As discussed in person, we're removing this.